### ISSUE TITLE
Issue in Setting the Precision

### ISSUE THREAD

**Issue by SirNate0** [mpmath/mpmath#842]
Could the [docs about setting the precision](https://mpmath.org/doc/current/basics.html#setting-the-precision) be updated to make it very clear that `mp` is an object in the mpmath module, and does not refer to the module itself (from `import mpmath as mp`)? At least two separate times I have attempted to increase the precision in my code by setting `mpmath.dps = 100` (not necessarily that value), thought I succeeded, and then only later realized that my results were all using the default dps of 15.

I don't know if it's possible given how Python handles assignment and all, but it would be even better if the library itself would issue a warning if mpmath.dps and the like were assigned to.

**Comment by oscarbenjamin** [comment]
There was a previous issue about this which even lead to a PEP proposal but it ultimately got rejected:

https://peps.python.org/pep-0726/

**Comment by WarrenWeckesser** [comment]
It has been a common problem; see https://github.com/mpmath/mpmath/issues/657

In mpmath 1.4.0, you'll get an error if you try to set `mpmath.dps`:

```
In [1]: import mpmath

In [2]: mpmath.__version__
Out[2]: '1.4.0a1'

In [3]: mpmath.dps = 200
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 mpmath.dps = 200

File ~/py3.12.4/lib/python3.12/site-packages/mpmath/__init__.py:449, in _MPMathModule._helper(self, prop, *args)
    448 def _helper(self, *args, prop=''):
--> 449     raise AttributeError("cannot set '{name}' on 'mpmath'. Did you mean to "
    450                          "set '{name}' on 'mpmath.mp'?".format(name=prop))

AttributeError: cannot set 'dps' on 'mpmath'. Did you mean to set 'dps' on 'mpmath.mp'?

```

**Comment by SirNate0** [comment]
Cool, I'm running 1.3.0 now, so I'm glad it's been fixed in the next version. Closing this, then, since it was already discussed extensively in the issue you linked.

**Comment by skirpichev** [comment]
On another hand, it's a recurring issue and I've no idea when 1.4 will be released (some issues I can't address without @fredrik-johansson).

Maybe it worth to release a 1.3.1 with bugfix for this issue?

**Comment by oscarbenjamin** [comment]
> some issues I can't address

Which issues?

Do they need to be addressed in 1.4?

**Comment by skirpichev** [comment]
> Which issues?

#709 at least

> Do they need to be addressed in 1.4?

Well, site should be updated to point on new stable release.  Ideally, it's good to do for a bugfix release too, especially if it will include documentation fixes.  But that's less severe.

**Comment by skirpichev** [comment]
Ok, it seems there are no big fans of this idea, closing.
